samples/zephyr_external_config/CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2022 Legrand North America, LLC.
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+# Add a common dts overlay necessary to ensure mcuboot is linked into,
+# and fits inside, the boot partition. (If the user specified a
+# DTC_OVERLAY_FILE on the CMake command line, we need to append onto
+# the list).
+if(DTC_OVERLAY_FILE)
+  set(DTC_OVERLAY_FILE
+	  "${DTC_OVERLAY_FILE} \${ZEPHYR_MCUBOOT_MODULE_DIR}/boot/zephyr/dts.overlay"
+    CACHE STRING "" FORCE
+    )
+else()
+  set(DTC_OVERLAY_FILE "\${ZEPHYR_MCUBOOT_MODULE_DIR}/boot/zephyr/dts.overlay")
+endif()
+
+
+# Enable Zephyr runner options which request mass erase if so
+# configured.
+#
+# Note that this also disables the default "leave" option when
+# targeting STM32 DfuSe devices with dfu-util, making the chip stay in
+# the bootloader after flashing.
+#
+# That's the right thing, because mcuboot has nothing to do since the
+# chip was just erased. The next thing the user is going to want to do
+# is flash the application. (Developers can reset DfuSE devices
+# manually to test mcuboot behavior on an otherwise erased flash
+# device.)
+macro(app_set_runner_args)
+  if(CONFIG_ZEPHYR_TRY_MASS_ERASE)
+    board_runner_args(dfu-util "--dfuse-modifiers=force:mass-erase")
+    board_runner_args(pyocd "--flash-opt=-e=chip")
+    board_runner_args(nrfjprog "--erase")
+  endif()
+endmacro()
+
+# find_package(Zephyr) in order to load application boilerplate:
+# http://docs.zephyrproject.org/application/application.html
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(mcuboot_exernal_config)
+
+target_sources(app PRIVATE
+  keys.c
+  )
+
+if(NOT CONFIG_BOOT_SIGNATURE_KEY_FILE STREQUAL "")
```

Comment:
Can we have a commit before this that refactors cmake in mcuboot a bit, so this code can be shared with the other cmake files instead of duplicated here. Perhaps `zephyr/cmake`.

Reply:
Done.